pkg/internal/client/txn.go

```diff
@@ -706,6 +707,11 @@ func (txn *Txn) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.
 	// If we're not yet writing in this txn, but intend to, insert a
 	// begin transaction request before the first write command.
 	if needBeginTxn {
+		// If we're setting the system-config trigger, anchor the transaction to
```

Comment:
Because they're separate concepts. I'd like to use `Txn.SetTxnKey` to make splitting, merging and change replicas less fragile.